src/components/filters/RandomFilter.jsx

```diff
@@ -2,47 +2,108 @@ import React from 'react';
 
 import FilterBase from './FilterBase';
 import IntInput from '../forms/inputs/IntInput';
+import SelectInput from '../forms/inputs/SelectInput';
+import Button from '../misc/Button';
 import makeRandomString from '../../utils/makeRandomString';
+import { injectIntl } from 'react-intl';
 
-
+@injectIntl
 export default class RandomFilter extends FilterBase {
     constructor(props) {
         super(props);
 
         this.state = {
-            size: props.config.size || 20,
+            amount: props.config.amount || 20,
             seed: props.config.seed || makeRandomString(6),
+            unit: props.config.unit || 'people'
         };
     }
 
     componentReceivedProps(nextProps) {
         if (nextProps.config !== this.props.config) {
             this.setState({
-                size: nextProps.config.size,
+                amount: nextProps.config.amount,
                 seed: nextProps.config.seed,
+                unit: nextProps.config.unit,
             });
         }
     }
 
     renderFilterForm(config) {
+        const { unit, amount } = this.state;
+        const msg = id => this.props.intl.formatMessage({ id });
+        const units = {
+            'people': msg('filters.random.units.people'),
+            'percentage': msg('filters.random.units.percentage'),
+        };
+
         return [
             <IntInput key="size" name="size"
+                className="RandomFilter__size"
                 labelMsg="filters.random.size"
-                value={ this.state.size }
-                onValueChange={ this.onSizeChange.bind(this) }
-                />,
+                value={ amount }
+                onValueChange={ this.onSizeChange }
+            />,
+            <SelectInput key="unit" name="unit"
+                className="RandomFilter__unit"
+                labelMsg="filters.random.units.label"
+                options={ units }
+                value={ this.state.unit }
+                onValueChange={ this.onUnitChange }
+            />,
+            <div className="RandomFilter__btnWrapper">
+                <Button
+                    labelMsg="filters.random.randomize"
+                    onClick={ this.onSeedChange }
+                />
+            </div>
         ];
     }
 
     getConfig() {
-        return {
-            size: this.state.size,
-            seed: this.state.seed,
-        };
+        const { amount, seed, unit } = this.state;
+        // We're doing this because percentages are interpeted
+        // as fractions in the backend.
+        const size = unit === 'percentage'
+            ? parseFloat('0.' + amount)
+            : amount;
+
+        return { size, amount, seed, unit };
```

Comment:
This should not return more than the properties allowed in the filter configuration that we send to the back-end, i.e. `seed` and `size`.

Instead of reyling on `amount` and `unit` being stored like this, derive them from the `size` in the constructor and `componentWillReceiveProps()`.